### PR TITLE
feat: Make create group optional

### PR DESCRIPTION
**Pull request by robinbowes**
## Description
Makes group creation optional

This PR supercedes #80.

## Motivation and Context
Consider a scenario where we have two security groups and each group needs to refer to the other:

- security group 1 - allow access from security group 2
- security group 2 - allow access from security group 1

It is not possible to use the module to manage the groups in this situation as there are circular dependencies. This can be avoided by creating the groups separately and then adding the rules.

## Breaking Changes
This change should be backwards compatible.

## How Has This Been Tested?
I've been using this code for around 2½ years :)

### CODE REVIEW

**Comment by robinbowes** [comment]
The failing check is fixed in #215

**Comment by antonbabenko** [comment]
Hi Robin! Could you please update an example code also to show this feature in action?

**Comment by robinbowes** [comment]
@antonbabenko Done!

File: main.tf [hunk 1]
```diff
@@ -13,16 +16,16 @@ locals {
 # Security group with name
 ##########################
 resource "aws_security_group" "this" {
-  count = var.create && false == var.use_name_prefix ? 1 : 0
+  count = var.create && var.create_group && !var.use_name_prefix ? 1 : 0
 
-  name                   = var.name
-  description            = var.description
+  name                   = local.name
```
Comment (antonbabenko) [inline main.tf:21]: These 2 new locals are not necessary. You can revert this change.

File: main.tf [hunk 2]
```diff
@@ -2,7 +2,10 @@
 # Get ID of created Security Group
 ##################################
 locals {
+  description = var.create_group ? var.description : "Not used"
+  name        = var.create_group ? var.name : "Not used"
   this_sg_id = concat(
+    [var.security_group_id],
```
Comment (antonbabenko) [inline main.tf:8]: `this_sg_id` will always contain the first value, which is going to be an existing security group or an empty string. Please fix and run other examples to confirm that this PR does not break existing functionality.

**Review by antonbabenko — CHANGES_REQUESTED** [review]
Please rebase or merge `master` into this one.

File: main.tf (hunk 1 above)
Comment (robinbowes) [reply main.tf]: Done

File: main.tf (hunk 2 above)
Comment (robinbowes) [reply main.tf]: Fixed

**Comment by robinbowes** [comment]
> Please rebase or merge `master` into this one.

Done.

I ran the complete, http, and rules-only examples against eu-central-1 and can confirm they all work perfectly.

**Comment by robinbowes** [comment]
Closing for #218
